WireDomain/Sources/WireDomain/Notifications/VerifyUserSessionUseCase.swift

```diff
@@ -31,26 +31,34 @@ struct VerifyUserSessionUseCase {
     enum Failure: Error {
         case coreDataMissingSharedContainer
         case coreDataMigrationRequired
+        case syncV2IsNotEnabled
         case unableToLoadStores(Error)
         case userUnauthenticated
     }
 
     // MARK: - Properties
 
+    private let journal: any JournalProtocol
     private let cookieStorage: any CookieStorageProtocol
     private let coreData: any CoreDataStackProtocol
 
     init(
+        journal: any JournalProtocol,
         cookieStorage: any CookieStorageProtocol,
         coreData: any CoreDataStackProtocol
     ) {
+        self.journal = journal
         self.cookieStorage = cookieStorage
         self.coreData = coreData
     }
 
     /// Ensures user is properly authenticated.
 
     func invoke() async throws {
+        guard journal[.isSyncV2Enabled] else {
+            throw Failure.syncV2IsNotEnabled
+        }
```

Comment:
This is to handle the scenario where the extension is started before the main app has had a chance to enable sync v2. In which case, we won't process any notifications.